libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestRequest.m

```diff
@@ -39,7 +39,7 @@ - (id)initWithMethod:(SFRestMethod)method serviceHostType:(SFSDKRestServiceHostT
         self.baseURL = baseURL;
         self.path = path;
         self.queryParams = [queryParams mutableCopy];
-        self.endpoint = kSFDefaultRestEndpoint;
+        self.endpoint = (hostType == SFSDKRestServiceHostTypeCustom)?@"":kSFDefaultRestEndpoint;
```

Comment:
`baseUrl` doesn't always have the information of whether an API should go against `loginUrl` or `instanceUrl`. I think we still need `hostType` for those special `loginUrl` URLs like the `userInfo` endpoint. I forget the exact specifics.

Reply:
Looking at `prepareRequestForSend`, it looks like if path starts with https then it is used the full url. In that case, base url, end point and host type don't matter. 

Maybe the method `customUrlRequestWithMethod` should just take one parameter: the full url (which would be fed into the path field of the SFRestRequest).

Also the comment for the method is misleading since it doesn't help the case where you want to hit a custom end point on salesforce (e.g. an api under `/services/apexrest`). For that case, a useful factory method would be one taking end point and path.

So in summary, useful new factory methods would be:
```objective-c
+ (instancetype)customUrlRequestWithMethod:(SFRestMethod)method fullURL:(NSString *)fullURL queryParams:(nullable NSDictionary<NSString*, id> *)queryParams;

+ (instancetype)customEndPointRequestWithMethod:(SFRestMethod)method endPoint:(NSString *)endPoint path:(NSString *)path queryParams:(nullable NSDictionary<NSString*, id> *)queryParams;
```